plugins/github-pull-requests-board/src/hooks/useUserRepositories.tsx

```diff
@@ -13,33 +13,51 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { useApi } from '@backstage/core-plugin-api';
+import { BackstageUserIdentity, useApi } from '@backstage/core-plugin-api';
 import {
   catalogApiRef,
-  humanizeEntityRef,
-  useEntity,
+  getEntityRelations,
 } from '@backstage/plugin-catalog-react';
 import { useCallback, useEffect, useState } from 'react';
-import { getProjectNameFromEntity } from '../utils/functions';
+import { getProjectNameFromEntity, parseEntityRef } from '../utils/functions';
+import { RELATION_MEMBER_OF } from '@backstage/catalog-model';
 
-export function useUserRepositories() {
-  const { entity: teamEntity } = useEntity();
+export const useUserRepositories = (
+  identity: BackstageUserIdentity,
+  allowedKinds?: string[],
+) => {
   const catalogApi = useApi(catalogApiRef);
   const [repositories, setRepositories] = useState<string[]>([]);
 
   const getRepositoriesNames = useCallback(async () => {
-    const entitiesList = await catalogApi.getEntities({
-      filter: {
-        'spec.owner': humanizeEntityRef(teamEntity, { defaultKind: 'group' }),
-      },
+    const userRef = parseEntityRef(identity.userEntityRef);
```

Comment:
`catalogApi.getEntityByRef` can actually take the ref string as-is without first parsing it to its constituent parts